client/my-sites/hosting/main.js

```diff
@@ -53,6 +54,9 @@ export default connect( state => {
 	const siteId = getSelectedSiteId( state );
 
 	return {
-		isDisabled: ! isSiteAutomatedTransfer( state, siteId ),
+		// We should never hit this case for the development flag, as it is handled on the redirect, but just as an extra layer...
+		isDisabled:
+			config.isEnabled( 'hosting/non-atomic-support' ) &&
+			! isSiteAutomatedTransfer( state, siteId ),
```

Comment:
I don't think we need this guard, at worst a site slips in that is not transferred in which case it'd be more confusing to see enabled buttons/CTA